src/lib.rs

```diff
@@ -90,18 +100,22 @@ pub fn run_program_with_custom_state(bin_path: &str, mut vm: VMState) -> (U256,
                     LogOpcode::TransientStorageRead => todo!(),
                     LogOpcode::TransientStorageWrite => todo!(),
                 },
-                Variant::FarCall(_) => todo!(),
+                Variant::FarCall(far_call_variant) => far_call(&mut vm, &far_call_variant),
+                // TODO: This is not how return works. Fix when we have calls between contracts
+                // hooked up.
+                // This is only to keep the context for tests
+                Variant::Ret(_) if vm.running_frames.len() > 1 => {
+                    vm.pop_frame();
+                }
```

Comment:
Isn't it better to add this if inside Ret?
```
Variant::Ret(_) => {
    if vm.running_frames.len() > 1 => {
        vm.pop_frame();
    } else {
        break
    }
}
```

Reply:
You're right, I'll change it.